compiler/lib/global_flow.ml

```diff
@@ -550,6 +565,7 @@ let f p =
     ; possibly_mutable
     ; known_cases = Hashtbl.create 16
     ; applied_functions = Hashtbl.create 16
+    ; fast = not (Config.Flag.effects ())
```

Comment:
Making`fast` an argument of `f` would give more freedom to the caller (and maybe allow to use `~fast:false` when using `-O3`)